pkg/kube/utils.go

```diff
@@ -171,3 +174,30 @@ func PVCContainsReadOnlyAccessMode(pvc *corev1.PersistentVolumeClaim) bool {
 
 	return false
 }
+
+// AddLabelsToPodOptions adds additional label selector to `PodOptions`.
+func AddLabelsToPodOptions(
+	options *PodOptions,
+	targetKey,
+	targetValue string,
+) {
+	if options.Labels == nil {
+		options.Labels = make(map[string]string)
+	}
+	options.Labels[targetKey] = targetValue
+}
+
+// ValidateLabelKeyIsPresentFromContext: This is a helper validation function used to validate the presence of a
+// label key. Result of this is used to add target label selector to the pod.
+func ValidateLabelKeyIsPresentFromContext(ctx context.Context, keyPrefix string) (bool, string) {
+	fields := field.FromContext(ctx)
+	if fields == nil {
+		return false, ""
+	}
+	for _, f := range fields.Fields() {
+		if strings.HasPrefix(f.Key(), keyPrefix) {
+			return true, f.Value().(string)
+		}
+	}
```

Comment:
I am sorry, I think I am missing something and that is creating confusion. Why do we need these checks? Do we not know the complete key name that is present in the context? Why do we need to match the prefix only?

Reply:
Correct. It should match the entire key and not prefix. 
Fixed.